cpp/src/io/reader.cpp

```diff
@@ -247,21 +253,29 @@ las::Points Reader::GetPointsWithinBox(const Box &box, double min_resolution)
     return out;
 }
 
-int32_t Reader::GetDepthAtResolution(double resolution) const
+int32_t Reader::GetDepthAtResolution(double resolution)
 {
+    // If query resolution is <=0 return the octree's max depth
     if (resolution <= 0)
-        throw std::runtime_error("Reader::GetDepthAtResolution: Query resolution must be greater than 0.");
+    {
+        int32_t max_depth = 0;
```

Comment:
initialize this to -1, as the "invalid" return

Reply:
this would only happen if the file was empty, so other queries that depend on this function won't have any nodes to iterate over, thus you wouldn't need to update the other queries with this value change.